frontend/src/api/LogApi.js

```diff
@@ -20,10 +22,36 @@ async function getLogs(endpoint) {
     });
 }
 
+async function getLogsOfStream(endpoint) {
+  return axios
+    .get(endpoint, getAuthorizationHeader())
+    .then((response) => {
+      return Promise.resolve(
+        response.data.map((log) => {
+          // log.serialNumber = sender, log.logEntity.serialNumber = receiver
+          return new StreamLogInfo(
+            log.logEntity.dateTime,
+            log.logEntity.level,
+            log.serialNumber,
+            log.logEntity.serialNumber,
+            log.logEntity.message
+          );
+        })
+      );
+    })
+    .catch(() => {
+      snackbar("error", "Failed to fetch stream logs");
+    });
+}
+
 export async function getAllLogs() {
   return getLogs(process.env.REACT_APP_LOG);
 }
 
 export async function getDeviceLogs(deviceSerialNumber) {
   return getLogs(`${process.env.REACT_APP_LOG}/${deviceSerialNumber}`);
 }
+
+export async function getStreamLogs(streamId) {
+  return getLogsOfStream(`${process.env.REACT_APP_STREAMLOGS}/${streamId}`);
+}
```

Comment:
why do you have two separate functions for this?

Reply:
I did it to kind of match the other two functions that shared the getLogs() function but I can change it if necessary. There wasn't really any reason for it.

Reply:
481c7bf

Reply:
481c7bf 1b904e1